Use DRF to create/update/stop stream recording workers/threads

Dhruv88 opened the pull request:
1. Added the REST API for the app.
2. Added test.py for unit testing of the APIs.
3. Added the basic serializer class for Stream model.
4. Created a separate folder to store the video streams.
5. Tweaks to views.py and recorder.py to handle cases where invalid stream url is given.

Fixes #9

Dhruv88 committed:
Added stop api and tests for create, update and delete

Dhruv88 committed:
Added comments to apis

Dhruv88 committed:
Added get api and Tests and serializer for Stream model

Dhruv88 committed:
Added a separate folder to store the recordings
Added safe_get to prevent DoesNotExist exception in put api
Added post delete to custom handler
Removed the initial tests folder instead only test.py file
Added case to recorder if url not found to stop recording
Added teardown to tests to delete the recordings created while testing

Dhruv88 committed:
Corrected Typo

Dhruv88 committed:
Added Exception Handling when invalid stream provided

Dhruv88 committed:
stream recording folder

Dhruv88 committed:
Small bug fix

Dhruv88 committed:
Merge branch 'master' into master

KShivendu commented on api/smart_cam/urls.py:
```diff
@@ -19,5 +19,6 @@
     path("admin/", admin.site.urls),
-    path("stream/", views.StreamAPI.as_view()),
+    path("stream/", views.StreamAPI.as_view(), name = "stream"),
+    path('stream/<int:stream_id>', views.StreamAPI.as_view(), name = "stream_with_id")
```
Rename `stream/` as `streams/`

KShivendu commented on api/smart_cam/views.py:
```diff
@@ -16,31 +19,66 @@
+
+    def get(self, request, stream_id=None):
+        if stream_id is None:
+            responseData = Stream.objects.all() 
```
Rename `responseData` as `streams`

KShivendu commented on api/smart_cam/views.py:
```diff
@@ -16,31 +19,66 @@
 
-        return JsonResponse({"id": stream.id, "url": stream.url})
+        streamIdAndUrl = {"id": stream.id, "url": stream.url}
+        return JsonResponse(streamIdAndUrl,status=status.HTTP_201_CREATED)
```
Use a serializer for this step.

KShivendu commented on api/smart_cam/views.py:
```diff
@@ -16,31 +19,66 @@
+    #safe_get to avoid Stream.DoesNotExist exception
+    def safe_get(self,id):
+        try:
+            got = Stream.objects.get(id=id)
```
rename `got` as `stream`

KShivendu commented on api/smart_cam/views.py:
```diff
@@ -16,31 +19,66 @@
+        stream = self.safe_get(id=stream_id)
+        updatedEnabledVal = json.loads(request.data["enabled"].lower())
+
+        if stream is not None:
```
variables should be name like python's standard guidelines (not JS)

`updatedEnabledVal` => `updated_enabled_val`
...
...

Dhruv88 committed:
1.) Made the suggested changes by KShivendu
2.) Added the create and update methods to the serializers
3.) Updated the tests according to updated urlpatterns
4.) Made the model's 'url' property unique to avoid duplicate streams
5.) Cleaned up the APi code in views.py
6.) Changed the valid attribute of Video_Recorder to is_valid and set 'enabled' to 'False' in stop_recording function in recorder.py

Dhruv88 committed:
Updated
